dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApproval.java

```diff
@@ -336,74 +337,17 @@ public boolean equals( Object object )
         {
             return true;
         }
-
-        if ( object == null || !(object instanceof DataApproval) )
+        if ( !(object instanceof DataApproval) )
         {
             return false;
         }
 
-        DataApproval that = (DataApproval) object;
-
-        if ( dataApprovalLevel != null )
-        {
-            if ( !dataApprovalLevel.equals( that.getDataApprovalLevel() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getDataApprovalLevel() != null )
-        {
-            return false;
-        }
-
-        if ( workflow != null )
-        {
-            if ( !workflow.equals( that.getWorkflow() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getWorkflow() != null )
-        {
-            return false;
-        }
-
-        if ( period != null )
-        {
-            if ( !period.equals( that.getPeriod() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getPeriod() != null )
-        {
-            return false;
-        }
-
-        if ( organisationUnit != null )
-        {
-            if ( !organisationUnit.equals( that.getOrganisationUnit() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getOrganisationUnit() != null )
-        {
-            return false;
-        }
-
-        if ( attributeOptionCombo != null )
-        {
-            if ( !attributeOptionCombo.equals( that.getAttributeOptionCombo() ) )
-            {
-                return false;
-            }
-        }
-        else if ( that.getAttributeOptionCombo() != null )
-        {
-            return false;
-        }
+        DataApproval other = (DataApproval) object;
 
-        return true;
+        return Objects.equals( dataApprovalLevel, other.dataApprovalLevel )
+            && Objects.equals( workflow, other.workflow )
+            && Objects.equals( period, other.period )
+            && Objects.equals( organisationUnit, other.organisationUnit )
+            && Objects.equals( attributeOptionCombo, other.attributeOptionCombo );
```

Comment:
Looks good. Should we use `Objects.hash` or `Objects.hashCode` for `hashCode` just to make it consistent and equally modern while at it?

Reply:
Definite proof it was too warm lately: I totally forgot `Objects.hash` existed :grin: